test/models/PostModel/test.js

```diff
@@ -0,0 +1,267 @@
+} = require('test/testHelper');
+const _ = require('lodash');
+const { PostFactory, wObjectFactory } = require('test/factories');
+
+describe('Post Model', async () => {
+  describe('On aggregate', async () => {
+    let post, author, permlink, title, body, postsCount;
+    beforeEach(async () => {
+      author = faker.name.firstName();
+      permlink = faker.random.string(20);
+      title = faker.address.city();
+      body = faker.lorem.sentence();
+      postsCount = faker.random.number(30);
+      for (let i = 0; i < postsCount; i++) {
+        if (i === 0) {
+          post = await PostFactory.Create({
+            author, permlink, title, body,
+          });
+        } else {
+          await PostFactory.Create({
+          });
+        }
+      }
+    });
+    describe('On group stage case', async () => {
+      let result;
+      beforeEach(async () => {
+        result = await PostModel.aggregate([{
+          $group: {
+            _id: '$_id',
+            author: { $first: '$author' },
+            permlink: { $first: '$permlink' },
+            title: { $first: '$title' },
+            body: { $first: '$body' },
+          },
+        }]);
+      });
+      it('Should return right count posts', async () => {
+        const { posts } = result;
+        expect(posts.length).to.be.eq(postsCount);
+      });
+      it('Should return records with id, author, perlink, title and body ', async () => {
+        const { posts: [posts] } = result;
+        expect(posts).to.have.all.keys(
+          '_id', 'author', 'permlink', 'title', 'body',
+        );
+      });
+    });
+
+    it('Should return from request: author, title and body', async () => {
+      const result = await PostModel.aggregate([{
+        $match: {
+          author: post.author,
+        },
+      },
+      {
+        $project: {
+          _id: 0, author: 1, title: 1, body: 1,
+        },
+      },
+      ]);
+      const { posts: [posts] } = result;
+      expect(posts).to.be.deep.eq(_.pick(post, ['author', 'title', 'body']));
+    });
+    it('Should check that the error message', async () => {
+      const { error } = await PostModel.aggregate([{
+        $match: {
+          author: faker.name.firstName(),
+        },
+      }]);
+      expect(error).to.be.exist;
+    });
+  });
+  describe('On getPostsRefs', async () => {
+    let skip, limit, postsCount;
+    beforeEach(async () => {
+      postsCount = faker.random.number(50);
+      for (let i = 0; i < postsCount; i++) {
+        await PostFactory.Create();
+      }
+      limit = faker.random.number(postsCount);
+      skip = faker.random.number(limit);
+    });
+    afterEach(async () => {
+      sinon.restore();
+    });
+    it('Should return right count records considering limits and skips', async () => {
+      const { posts } = await PostModel.getPostsRefs({ skip, limit });
+      const answer = postsCount - skip >= limit ? limit : postsCount - skip;
```

Comment:
const answer = (postsCount - skip) >= limit ? limit : (postsCount - skip); what would be a little clearer

Reply:
only two cases for this method?  check at least that all keys returned everything